tests/test_sphinx.py

```diff
@@ -8,25 +8,38 @@
 
 import pytest
 
+from collections.abc import Mapping
+
 from sphinx.application import Sphinx
 
 from rinoh.document import DocumentTree
-from rinoh.frontend.sphinx import template_from_config
+from rinoh.frontend.sphinx import template_from_config, set_document_metadata
 from rinoh.language import IT
 from rinoh.paper import A4, LETTER
 from rinoh.templates import Book, Article
+from rinoh.template import DocumentTemplate
+
+from docutils.frontend import OptionParser
 
 
 def create_sphinx_app(tmpdir, **confoverrides):
-    return Sphinx(srcdir=tmpdir.strpath,
+    app = Sphinx(srcdir=tmpdir.strpath,
                   confdir=None,
                   outdir=(tmpdir / 'output').strpath,
                   doctreedir=(tmpdir / 'doctrees').strpath,
                   buildername='rinoh',
                   # confoverrides=dict(extensions=['rinoh.frontend.sphinx'],
                   #                    **confoverrides))
                   confoverrides=confoverrides)
+    return app
 
+def create_doctree(title="A Title", author="Ann Other", docname="a_name"):
+    doctree = DocumentTree([])
+    doctree.settings = OptionParser().get_default_values()
```

Comment:
A rinohtype DocumentTree instance does not have a settings attribute. A docutils doctree does, however. I see you addresses this mistake in your last commit, but this test still needs to be fixed.

Reply:
Thanks for the hint, updated the tests to use docutils doctree.